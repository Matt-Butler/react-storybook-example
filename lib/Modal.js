'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('../src/Modal.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Modal = function Modal(props) {
  return _react2.default.createElement(
    'h1',
    { className: 'terra-Modal' },
    'Hello from ',
    props.phrase,
    '!'
  );
};

Modal.propTypes = {
  phrase: _react2.default.PropTypes.string.isRequired
};

Modal.defaultProps = {
  phrase: 'a Stranger'
};

exports.default = Modal;