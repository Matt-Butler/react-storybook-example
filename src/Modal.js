import React, { Component, PropTypes } from 'react';
// import '../src/Modal.scss';

class Modal extends Component {
  render() {
    return <h1 className="terra-Modal">Hello from {this.props.phrase}!</h1>;
  }
}

export default Modal;
