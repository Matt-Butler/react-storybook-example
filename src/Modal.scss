.terra-Modal {
  background-color: #EEE;
  color: #000;
}
