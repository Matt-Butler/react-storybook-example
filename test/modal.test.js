import React from 'react';
import Modal from '../src/modal.js';

it('should shallow my modal', () => {
    const wrapper = shallow(
        <Modal phrase="Jest" />
    );
    expect(wrapper).toMatchSnapshot();
});

it('should mount my modal', () => {
    const wrapper = mount(
        <Modal phrase="Jest" />
    );
    expect(wrapper).toMatchSnapshot();
});


it('should render my modal', () => {
    const wrapper = render(
        <Modal phrase="Jest" />
    );
    expect(wrapper).toMatchSnapshot();
});
