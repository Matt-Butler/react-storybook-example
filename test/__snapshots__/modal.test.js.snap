exports[`Storyshots Modal With phrase L5 1`] = `
<h1
  className="terra-Modal">
  Hello from 
  L5
  !
</h1>
`;

exports[`Storyshots Modal With phrase Matt 1`] = `
<h1
  className="terra-Modal">
  Hello from 
  Matt
  !
</h1>
`;

exports[`test should mount my modal 1`] = `
<Modal
  phrase="Jest">
  <h1
    className="terra-Modal">
    Hello from 
    Jest
    !
  </h1>
</Modal>
`;

exports[`test should render my modal 1`] = `
<h1
  class="terra-Modal">
  Hello from Jest!
</h1>
`;

exports[`test should shallow my modal 1`] = `
<h1
  className="terra-Modal">
  Hello from 
  Jest
  !
</h1>
`;
